cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/edges/flows/ProgramDependence.kt

```diff
@@ -70,8 +69,32 @@ class ProgramDependences<NodeType : Node> :
     }
 
     override fun add(e: Edge<NodeType>): Boolean {
-        // Clone the edge before inserting. See comment above for a detailed explanation.
-        val clonedEdge = e.clone()
-        return super<EdgeSet>.add(clonedEdge)
+        return super<EdgeSet>.add(e)
+    }
+}
+
+/**
+ * This edge class defines that there's some kind of dependency between [start] and [end]. The
+ * nature of this dependency is defined by [dependence].
+ */
+@RelationshipEntity
+open class ProgramDependence(start: Node, end: Node, dependence: DependenceType) :
+    Edge<Node>(start, end) {
+    init {
+        this.dependence = dependence
```

Comment:
would it make sense to move the dependence field to this class instead of having it in every edge?

Reply:
I guess so because I don't know what should be the use-case of having it in the `Edge` class in general. I'll update this.